aws/resource_aws_codebuild_project.go

```diff
@@ -659,6 +679,22 @@ func expandProjectEnvironment(d *schema.ResourceData) *codebuild.ProjectEnvironm
 		projectEnv.ImagePullCredentialsType = aws.String(v.(string))
 	}
 
+	if v := envConfig["registry_credential"]; v != nil {
```

Comment:
It looks like the code around this might not be performing the safer check, but could you please update this to something like the following to perform the existence checking? Thanks!

```suggestion
	if v, ok := envConfig["registry_credential"]; ok && len(v.([]interface{})) > 0 {
```